ext/opentelemetry-ext-django/src/opentelemetry/ext/django/middleware.py

```diff
@@ -0,0 +1,110 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from logging import getLogger
+
+from django.utils.deprecation import MiddlewareMixin
+
+from opentelemetry.context import attach, detach
+from opentelemetry.ext.django.version import __version__
+from opentelemetry.ext.wsgi import (
+    add_response_attributes,
+    collect_request_attributes,
+    get_header_from_environ,
+)
+from opentelemetry.propagators import extract
+from opentelemetry.trace import SpanKind, get_tracer
+
+_logger = getLogger(__name__)
+
+
+class OpenTelemetryMiddleware(MiddlewareMixin):
+    """Django Middleware for OpenTelemetry
+    """
+
+    _environ_activation_key = (
+        "opentelemetry-instrumentor-django.activation_key"
+    )
+    _environ_token = "opentelemetry-instrumentor-django.token"
+    _environ_span_key = "opentelemetry-instrumentor-django.span_key"
+
+    def process_view(
+        self, request, view_func, view_args, view_kwargs
+    ):  # pylint: disable=unused-argument
+        # request.META is a dictionary containing all available HTTP headers
+        # Read more about request.META here:
+        # https://docs.djangoproject.com/en/3.0/ref/request-response/#django.http.HttpRequest.META
+
+        # environ = {
+        #     key.lower().replace('_', '-').replace("http-", "", 1): value
+        #     for key, value in request.META.items()
+        # }
+
+        environ = request.META
```

Comment:
Can request.headers be used instead?

Reply:
Tried to, got this error while running the test cases:

```
___________________________________________________________________ TestMiddleware.test_traced_post ____________________________________________________________________
                                                                                                                                                                        
self = <tests.test_middleware.TestMiddleware testMethod=test_traced_post>                                                                                               
                                                                                                                                                                        
    def test_traced_post(self):                                                                                                                                         
>       Client().post("/traced/")                                                                                                                                       
                                                                                                                                                                        
test_middleware.py:72:                                                                                                                                                  
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../.tox/py38-test-ext-django/lib/python3.8/site-packages/django/test/client.py:526: in post                                                                       
    response = super().post(path, data=data, content_type=content_type, secure=secure, **extra)                                                                         
../../../.tox/py38-test-ext-django/lib/python3.8/site-packages/django/test/client.py:355: in post                                                                       
    return self.generic('POST', path, post_data, content_type,                                                                                                          
../../../.tox/py38-test-ext-django/lib/python3.8/site-packages/django/test/client.py:421: in generic                                                                    
    return self.request(**r)                                                                                                                                            
../../../.tox/py38-test-ext-django/lib/python3.8/site-packages/django/test/client.py:496: in request                                                                    
    raise exc_value                                                                                                                                                     
../../../.tox/py38-test-ext-django/lib/python3.8/site-packages/django/core/handlers/exception.py:34: in inner                                                           
    response = get_response(request)                                                                                                                                   
../../../.tox/py38-test-ext-django/lib/python3.8/site-packages/django/core/handlers/base.py:106: in _get_response                                                       
    response = middleware_method(request, callback, callback_args, callback_kwargs)                                                                                     
../../../.tox/py38-test-ext-django/lib/python3.8/site-packages/opentelemetry/ext/django/middleware.py:60: in process_view                                               
    attributes = collect_request_attributes(environ)                                                                                                                    
../../../.tox/py38-test-ext-django/lib/python3.8/site-packages/opentelemetry/ext/wsgi/__init__.py:127: in collect_request_attributes                                    
    "http.method": environ["REQUEST_METHOD"],                                                                                                                           
../../../.tox/py38-test-ext-django/lib/python3.8/site-packages/django/http/request.py:388: in __getitem__
    return super().__getitem__(key.replace('_', '-'))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = {'Cookie': '', 'Content-Length': '20', 'Content-Type': 'multipart/form-data; boundary=BoUnDaRyStRiNg'}, key = 'REQUEST-METHOD'

    def __getitem__(self, key):
>       return self._store[key.lower()][1]
E       KeyError: 'request-method'

../../../.tox/py38-test-ext-django/lib/python3.8/site-packages/django/utils/datastructures.py:320: KeyError
```